### PR TITLE
Extended validation functionality to allow a range of values and multiple values to be considered valid

### PR DESCRIPTION
**Pull request by daniellesilfa**
This PR consists of code that extends the functionality of the validation to allow for validation using a range of values and a set of allowed values for settings. More schema interfaces were added to schema.ts to increase the amount of available schemas types that can be used for creating the resulting schema object. The scd.ts file was modified to use the new schema interfaces that were created instead of an EnumSchema object for all of the string values. In addition, the validator class was modified to perform checks using the new schema types created.

### CODE REVIEW

File: server/src/common/schema.ts [hunk 1]
```diff
@@ -43,6 +45,21 @@ export interface MapSchema {
   };
 }
 
+export interface RangeSchema {
+  type: 'range'; 
+  valueSchema: 'string'; 
+  rangeValues: {
+  lowvalue: number;
```
Comment (robertschaedler3) [inline server/src/common/schema.ts:52]: nit: indentation/formatting

File: server/src/common/schema.ts [hunk 2]
```diff
@@ -104,6 +121,22 @@ export const MapSchema: D.Decoder<unknown, MapSchema> = D.struct({
   }),
 });
 
+export const RangeSchema: D.Decoder<unknown, RangeSchema> = D.struct({
+  type: D.literal('range'),
+  //name: D.string,
```
Comment (robertschaedler3) [inline server/src/common/schema.ts:126]: nit: extra comment

File: server/src/common/schema.ts [hunk 3]
```diff
@@ -214,6 +257,8 @@ export abstract class Visitor<Context> {
   public abstract onArray(node: Node, context?: Context): void;
   public abstract onObject(node: Node, context?: Context): void;
   public abstract onProperty(node: Node, context?: Context): void;
+  //public abstract onRange(node: Node, context?: Context): void;
```
Comment (robertschaedler3) [inline server/src/common/schema.ts:260]: nit: extra comment (can this be removed?)

File: .github/workflows/ci.yml
Comment (robertschaedler3) [inline .github/workflows/ci.yml:1]: Weren't these changes this already checked in?

**Review by robertschaedler3 — DISMISSED** [review]

File: .github/workflows/ci.yml
Comment (daniellesilfa) [reply .github/workflows/ci.yml:1]: Yes, I'm not sure why/how I ended up checking them in again.

**Review by robertschaedler3 — APPROVED** [review]

**Pull request merged**